docs/testsuite_overview.md

```diff
@@ -0,0 +1,269 @@
+    * A ```manifest.json``` file containing a listing of all the test files to help drive automated test execution.
+    * Guidelines on how to use the spec to validate a Rest.li client implementation.
+* A Java TestNG suite that uses the spec's test data to validate the Java Rest.li client implementation.
+* A Rest.li Java server for code generation
+
+
+The test suite is structured as follows: 
+```
+.
+|—— client-testsuite
+|   |—— manifest.json
+|   |—— data/
+|   |—— requests/
+|   |—— requests-v2/
+|   |—— responses/
+|   |—— responses-v2/
+|   |—— restspecs/
+|   |—— schemas/
+|   |—— snapshots/
+|   |—— src/
+|   |—— ...
+|—— restli-testsuite-server
+|   |—— src
+|   |—— ...
+|—— build.gradle
+|... 
+```
+
+
+#### Rest.li Test Suite Specification
+
+##### manifest.json
+This manifest file provides machine readable information about all the automated tests included in this spec.
+It is shared across languages, and does not need to be changed when adding a test suite for a new language. 
+It should be changed when expanding the Rest.li Test Suite Specification.
+
+This file is intended to help drive portions of the test suite execution for each language. Although it does help,
+writing additional assertions by hand will still be needed in most languages to validate correctness of in-memory
+representations and language bindings.
+
+The file is broken down into a few main sections:
+
+* jsonTestData - list of JSON data files.
+* schemaTestData - list of Rest.li data schema files (.pdsc) as well as JSON data files matching the schemas.
+* wireProtocolTestData - list of Rest.li interface definition files (.restspec.json) as well as test HTTP requests and responses,
+  in the form of files, for operations supported by the interface definition.
+
+##### Test data folders
+These folders contain testing data used by the automated tests in the spec. Most of them were generated by Java Rest.li,
+and should be used when following the spec to test your Rest.li implementation. 
+
+The following folders are included:
+* **data**: input JSON data for testing schemas and json serialization
+* **requests**: correct HTTP requests for wire protocol tests (Rest.li protocol 1)
+* **requests-v2**: correct HTTP requests for wire protocol tests (Rest.li protocol 2)
+* **responses**: input HTTP responses for wire protocol tests (Rest.li protocol 1)
+* **responses-v2**: input HTTP responses for wire protocol tests (Rest.li protocol 2)
+* **restspecs**: restspecs generated from Java resources. These should be used by Rest.li implementations
+to make request builders.
+* **snapshots**: snapshots for resources
+
+#### Java TestNG Tests
+In ```src/test/java```, you can find the Java test suite, which uses the TestNG testing framework. 
+You can add test suites for new languages in ```src/test/```. The Java suite is separated into two folders:
+* **test**: Java files containing the tests, and utility methods for running tests. 
```

Comment:
We can simply this a bit

containing the tests (```TestRestClientAgainstStandardTestSuite``` and ```TestRestClientWithManualAssertions```), and utility methods (```StandardTestSuiteBase```)...